### ISSUE TITLE
[jdbi-generator] Java interface to work with @GenerateSqlObject must extends SqlObject

### ISSUE THREAD

**Issue by wbprime** [jdbi/jdbi#1791]
Given following code, jdbi-generator generates correct impl class of JdbiDao interface:

```
@GenerateSqlObject
public interface JdbiDao extends SqlObject {
    @SqlUpdate("insert into users (name) values ( ?)")
    @GetGeneratedKeys("id")
    long insert(String name);
}
```

However for:

```
@GenerateSqlObject
public interface JdbiDao {
    @SqlUpdate("insert into users (name) values ( ?)")
    @GetGeneratedKeys("id")
    long insert(String name);
}
```

`mvn compile` complains :

> [ERROR]   JdbiDaoImpl.java:[71,5] method does not override or implement a method from a supertype
> [ERROR]   JdbiDaoImpl.java:[73,78] cannot find symbol
> [ERROR]   symbol:   method getHandle()
> [ERROR]   location: variable e of type com.example.JdbiDao

The problem locates in generated `JdbiDaoImpl.java` file, looking like:

```
public static class OnDemand implements JdbiDao {
    private final Jdbi db;

    public OnDemand(Jdbi db) {
      this.db = db;
    }

    @Override
    public long insert(String name) {
      return (long) db.withExtension(JdbiDao.class, e -> e.insert(name));
    }

    @Override
    public Handle getHandle() {
      return (org.jdbi.v3.core.Handle) db.withExtension(JdbiDao.class, e -> e.getHandle());
    }

    @Override
    public <R, X extends Exception> R withHandle(HandleCallback<R, X> callback) throws X {
      return (R) db.withExtension(JdbiDao.class, e -> e.withHandle(callback));
    }
  }
```

The generated `OnDemand` class is not a subclass/implementation of `SqlObject`, so it failes when compiling the method `getHandle` & `withHandle`.

So is this the expected behavior as designed ?

Maybe a generator direction could be added to supress the generation of `OnDemand` class (add a member in annotation GenerateSqlObject ?).

**Comment by stevenschlansker** [comment]
Thanks for reporting this, it is fixed in #1788 which will be in the next release

**Comment by wbprime** [comment]
Got it.

**Comment by stevenschlansker** [comment]
Fixed in 3.18.0
